File: code/PyWrapper_TKO/py_wrapper/UtilityList.py
from .Element import Element

class UtilityList :

	def __init__(self,item,sumIutils=0,sumRutils=0,elements =[]):
		"""

		:param item:
		:param elements:
		:param sumIutils:
		:param sumRutils:
		"""
		self.item = item
		self.elements = elements
		self.sumIutils = sumIutils
		self.sumRutils = sumRutils

	def addElement(self,element):
		"""
		pass an Element object
		"""
		self.sumIutils += element.iutils
		self.sumRutils += element.rutils
		self.elements.append(element)

File: code/TKO_CSharp/Models/UtilityList.cs
using System.Collections.Generic;

namespace TKO_CSharp.Models
{
    public class UtilityList
    {
        public int item;  // the item
        public int sumIutils = 0;  // the sum of item utilities
        public int sumRutils = 0;  // the sum of remaining utilities
        public List<Element> elements = new List<Element>();  // the elements

        public UtilityList(int item)
        {
            this.item = item;
        }

        /**
		 * Method to add an element to this utility list and update the sums at the same time.
		 */
        public void addElement(Element element)
        {
            sumIutils += element.iutils;
            sumRutils += element.rutils;
            elements.Add(element);
        }
    }

}